src/handle/root.rs

```diff
@@ -43,6 +43,21 @@ impl<T: Object> Root<T> {
     /// The caller _must_ ensure `Root::into_inner` or `Root::drop` is called
     /// to properly dispose of the `Root<T>`. If the value is dropped without
     /// calling one of these methods, it will *panic*.
+    ///
+    /// Be careful that you aren't short-circuiting with an early return before
+    /// your Root value gets into_inner'd or dropped. The following will cause
+    /// a runtime panic when your error case gets triggered:
+    /// ```
```

Comment:
This is failing doc tests for me locally. I'm not sure why it's not failing in CI. 😕

Reply:
I think I have fixed this, I wasn't even thinking about doc tests when I wrote the code.